File: Problems & Solutions/1833. Maximum Ice Cream Bars/06 January 2022/Approach 1 - Sorting (Greedy)/Maximum Ice Cream Bars.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        sort(costs.begin(), costs.end());

        int n = costs.size();
        int icecream = 0;

        while (icecream < n && costs[icecream] <= coins) {
            coins -= costs[icecream];
            icecream += 1;
        }

        return icecream;
    }
};

File: Problems & Solutions/1833. Maximum Ice Cream Bars/06 January 2022/Approach 1 - Sorting (Greedy)/Maximum Ice Cream Bars.swift
class Solution {
    func maxIceCream(_ costsArray: [Int], _ totalCoins: Int) -> Int {
        var coins = totalCoins
        var costs = costsArray

        costs.sort()

        let n: Int = costs.count
        var icecream: Int = 0

        while icecream < n && costs[icecream] <= coins {
            coins -= costs[icecream]
            icecream += 1
        }

        return icecream
    }
}